javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/config/ResourceProviderPropertiesCustomizer.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.tooling.config;
+
+import com.google.auto.service.AutoService;
+import io.opentelemetry.sdk.autoconfigure.spi.AutoConfigurationCustomizer;
+import io.opentelemetry.sdk.autoconfigure.spi.AutoConfigurationCustomizerProvider;
+import io.opentelemetry.sdk.autoconfigure.spi.ConfigProperties;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import javax.annotation.Nullable;
+
+@AutoService(AutoConfigurationCustomizerProvider.class)
+public class ResourceProviderPropertiesCustomizer implements AutoConfigurationCustomizerProvider {
+
+  private static final Set<String> DISABLED_BY_DEFAULT_RESOURCE_PROVIDERS =
+      new HashSet<>(
+          Arrays.asList(
+              "io.opentelemetry.contrib.aws.resource.BeanstalkResourceProvider",
```

Comment:
I think not all of these need to wait for http requests to timeout or respond. Should we by default disable only the slow ones? I guess for start it doesn't matter.

Reply:
My idea was to do that in a follow up PR - to keep this PR more about the mechanism.

Reply:
make sense